finality-aleph/src/tcp_network.rs

```diff
@@ -66,6 +70,28 @@ impl Listener for TcpListener {
 
 impl PeerId for AuthorityId {}
 
+impl PublicKey for AuthorityId {
+    type Signature = Signature;
+
+    fn verify(&self, message: &[u8], signature: &Self::Signature) -> bool {
+        verify(self, message, signature)
+    }
+}
+
+#[async_trait::async_trait]
+impl PrivateKey for AuthorityPen {
+    type Signature = Signature;
+    type PublicKey = AuthorityId;
+
+    async fn sign(&self, message: &[u8]) -> Self::Signature {
+        AuthorityPen::sign(self, message).await
```

Comment:
Can't it be just `self.sign(message).await`?

Reply:
[Ah, you fell for one of the classic blunders](https://www.youtube.com/watch?v=J3Z4yW60I2w)! I'll let you figure out why by yourself. :3

Reply:
Ahhh, yes...